libs/ConfigKit/include/ConfigKit.h

```diff
@@ -7,15 +7,38 @@
 #include <filesystem>
 
 #include "Config.h"
+#include "FileManagerKit.h"
 
 namespace leka {
 
 class ConfigKit
 {
   public:
 	explicit ConfigKit() = default;
-	[[nodiscard]] auto read(Config const &config) const -> uint8_t;
-	[[nodiscard]] auto write(Config const &config, uint8_t data) const -> bool;
+	template <std::size_t N>
+	[[nodiscard]] auto read(Config<N> const &config) const -> std::array<uint8_t, N>
+	{
+		if (FileManagerKit::File file {config.path(), "r"}; file.is_open()) {
+			auto data = std::array<uint8_t, N> {};
+			file.read(data);
+
+			return data;
+		}
+
+		return config.default_value();
+	}
```

Comment:
it's actually possible to have a function with different return types provided it's used in `constexpr` context

see https://gcc.godbolt.org/z/ren7Pa685

not sure if that would work there though, maybe for future improvements

Reply:
I tried something for read